src/vss_tools/exporters/graphql.py

```diff
@@ -1,134 +1,532 @@
+from vss_tools.utils.misc import getattr_nn
 
 
-class GraphQLFieldException(Exception):
+class GraphQLExporterException(Exception):
+    """Exception raised for errors in the GraphQL export process."""
+
     pass
 
 
-def get_schema_from_tree(root: VSSNode, additional_leaf_fields: list) -> str:
-    """Takes a VSSNode and additional fields for the leafs. Returns a graphql schema as string."""
-    args = dict(
-        id=GraphQLArgument(
-            GraphQLNonNull(GraphQLString),
-            description="VIN of the vehicle that you want to request data for.",
-        ),
-        after=GraphQLArgument(
-            GraphQLString,
-            description=(
-                "Filter data to only provide information that was sent " "from the vehicle after that timestamp."
-            ),
-        ),
-    )
+# ========= Custom GraphQL scalar types =========
+class Int8(Scalar):
+    pass
 
-    root_query = GraphQLObjectType(
-        "Query",
-        lambda: {"vehicle": GraphQLField(to_gql_type(root, additional_leaf_fields), args)},
-    )
-    return print_schema(GraphQLSchema(root_query))
 
+class UInt8(Scalar):
+    pass
+
+
+class Int16(Scalar):
+    pass
+
+
+class UInt16(Scalar):
+    pass
+
+
+class UInt32(Scalar):
+    pass
+
+
+class Int64(Scalar):
+    pass
+
+
+class UInt64(Scalar):
+    pass
+
+
+# ========= Mapping aids =========
+class GQLElementType(Enum):
+    """Enum of GraphQL elements to better handle naming conventions during the export process"""
+
+    TYPE = "type"
+    FIELD = "field"
+    ARGUMENT = "argument"
+    DIRECTIVE = "directive"
+    ENUM = "enum"
+    INTERFACE = "interface"
+    UNION = "union"
+    SCALAR = "scalar"
+    ENUM_VALUE = "enum_value"
+
+
+datatype_map = {
+    Datatypes.INT8[0]: Int8,
+    Datatypes.INT8_ARRAY[0]: graphene.List(Int8),
+    Datatypes.UINT8[0]: UInt8,
+    Datatypes.UINT8_ARRAY[0]: graphene.List(UInt8),
+    Datatypes.INT16[0]: Int16,
+    Datatypes.INT16_ARRAY[0]: graphene.List(Int16),
+    Datatypes.UINT16[0]: UInt16,
+    Datatypes.UINT16_ARRAY[0]: graphene.List(UInt16),
+    Datatypes.INT32[0]: graphene.Int,
+    Datatypes.INT32_ARRAY[0]: graphene.List(graphene.Int),
+    Datatypes.UINT32[0]: UInt32,
+    Datatypes.UINT32_ARRAY[0]: graphene.List(UInt32),
+    Datatypes.INT64[0]: Int64,
+    Datatypes.INT64_ARRAY[0]: graphene.List(Int64),
+    Datatypes.UINT64[0]: UInt64,
+    Datatypes.UINT64_ARRAY[0]: graphene.List(UInt64),
+    Datatypes.FLOAT[0]: graphene.Float,
+    Datatypes.FLOAT_ARRAY[0]: graphene.List(graphene.Float),
+    Datatypes.DOUBLE[0]: graphene.Float,
+    Datatypes.DOUBLE_ARRAY[0]: graphene.List(graphene.Float),
+    Datatypes.BOOLEAN[0]: graphene.Boolean,
+    Datatypes.BOOLEAN_ARRAY[0]: graphene.List(graphene.Boolean),
+    Datatypes.STRING[0]: graphene.String,
+    Datatypes.STRING_ARRAY[0]: graphene.List(graphene.String),
+}
+
+# ========= Global variables =========
+vss_branches_df = pd.DataFrame()
+vss_leaves_df = pd.DataFrame()
```

Comment:
should be `vss_leafs` instead of `leaves`. I do not like encoding the datatype into the name `_df`

Reply:
You are right, my bad

Reply:
Or well, looked it up. Both variants are valid